crates/fuel-core/src/graphql_api/api_service.rs

```diff
@@ -214,6 +226,66 @@ impl RunnableTask for Task {
     }
 }
 
+const REQUIRED_FUEL_BLOCK_HEIGHT_HEADER: &str = "REQUIRED_FUEL_BLOCK_HEIGHT";
+const CURRENT_FUEL_BLOCK_HEIGHT_HEADER: &str = "CURRENT_FUEL_BLOCK_HEIGHT";
+
+async fn required_fuel_block_height<B>(
+    req: AxumRequest<B>,
+    next: Next<B>,
+) -> impl IntoResponse {
```

Comment:
Nit: I prefer more informative names for generics if they don't have any clear trait bounds:
```suggestion
async fn required_fuel_block_height<Body>(
    req: AxumRequest<Body>,
    next: Next<Body>,
) -> impl IntoResponse {
```